tests/Feature/Api/CancelOrderTest.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Tests\Feature\Api;
+
+use Tests\TestCase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Illuminate\Http\Response;
+use App\User;
+use App\Order;
+
+class CancelOrderTest extends TestCase
+{
+    use DatabaseMigrations;
+    /**
+     * Set up order
+     *
+     * @return void
+     */
+    public function setUp()
+    {
+        parent::setUp();
+        factory(Order::class)->create([
```

Comment:
tạo 1 biến local:
protected $order;
$this->order = factory(Order::class)->create(....)
// không cần phải query ngược vào db để lấy ra 
// no need: $order = Order::first();
$this->jsonUser('PUT', '/api/orders/'. $this->order->id .'/cancel', $cancel);